indra/tools/reading/submit_reading_pipeline_aws.py

```diff
@@ -86,6 +94,17 @@ def check_logs(job_defs):
                 old_log += log_lines[len(old_log):]
         return stalled_jobs
 
+    # Don't start watching jobs added after this command was initialized.
+    observed_job_def_set = set()
+
+    if stash_log_method is not None:
+        def update_observed_jobs(job_defs):
+            for job_def in job_defs:
+                observed_job_def_set.add(
+                    tuple([(k, v) for k, v in job_def.items()
+                           if k in ['jobName', 'jobId']])
```

Comment:
I'm trying to wrap my head around this, it might be clearer to do 
```
# Build up a set of tuples representing job names and ids for each job to add
jobs_to_add = {(('jobName', job_def['jobName']), 
                ('jobId', job_def['jobId'])) for job_def in job_defs}
# Update the observed jobs
observed_job_def_set |= jobs_to_add
```

Reply:
That is so that if a job meeting the criteria comes up after `wait_for_complete` has started. The idea is that anything the script "waits for" gets their logs stashed.